### ISSUE TITLE
Invalid selectors cause application to crash

### ISSUE THREAD

**Issue by janhoogeveen** [modulz/stitches#626]
# Bug report

## Describe the bug

My application was crashing with the following errors:

Firefox: `DOMException: An invalid or illegal string was specified`
Chrome: `VM78 react_devtools_backend.js:2560 DOMException: Failed to execute 'insertRule' on 'CSSGroupingRule': the rule '@hoverNone{.c-imFVOV{white-space:nowrap;overflow-x:scroll;padding:var(--space-2) 0}}' is invalid and cannot be parsed`
Safari: `SyntaxError: The string did not match the expected pattern.`

## To Reproduce

1. Define a theme

```js
export const { theme, css, styled, global, getCssString } = createCss({
  theme: {
    colors: {
      background: 'white',
      foreground: 'black',
    },
    space: {
      0: '0',
      1: '0.25rem',
    },
  },
  media: {
    bp1: '(min-width: 40em)',
    bp2: '(min-width: 52.5em)',
    noHover: '(hover: none)',
  },
});
```

2. Create a styled component, but use a non-existing media query name

In this example I use `@hoverNone`, but the right name should be `@noHover`. 

```js
const Subcategories = styled('div', {
  '@bp2': {
    whiteSpace: 'normal',
  },
  '@hoverNone': {
    whiteSpace: 'nowrap',
    overflowX: 'scroll',
    padding: '$2 0',
  }
});
```

3. It will crash
4. Change `@hoverNone` to `@noHover` and things are OK.

I found another instance in my application which crashed because we used `$ > div` instead of `& > div` in a nested selector.

## Expected behavior

Catch the error before the application crashes, and log a warning to the console that some styles could not be applied because you wrote an invalid selector.

## System information

- OS: `macOS`
- Browser: `Firefox 89/Chrome 91/Safari 14.1`
- Version of Stitches: `"@stitches/react": "^0.2.0-canary.4"`
- Version of `Node.js: 14.x on Vercel`

**Comment by peduarte** [comment]
Thanks for that. Repro: https://codesandbox.io/s/zen-shadow-xpkce?file=/src/App.tsx

**Comment by norman-ags** [comment]
Also happening when adding breakpoints in `css` prop. Repro: https://codesandbox.io/s/kind-flower-mg04z?file=/src/App.tsx

**Comment by peduarte** [comment]
Another repro: https://codesandbox.io/s/sparkling-http-69zg6?file=/src/components/Playground/index.tsx:342-355

**Comment by peduarte** [comment]
Fixed, will be released in `0.2.1`.

Demos:
- https://codesandbox.io/s/626-pr-iw7rn
- https://codesandbox.io/s/626-pr-norman-hzg40
- https://codesandbox.io/s/626-pr-fixed-3p1uy

**Comment by peduarte** [comment]
This is now available in the latest version

```bash
npm i @stitches/react@latest
```
